### ISSUE TITLE
HCL AppScan - Cryptography.InsecureAlgorithm found in modelmapper code

### ISSUE THREAD

**Issue by troy-bishop** [modelmapper/modelmapper#640]
Hello,

I am a developer for an IBM product that utilizes and packages modelmapper.  One of our requirements is to run various security scanning tools against our product and the security scanning tool (HCL AppScan) found the following insecure algorithm usage:

```
Location: org.modelmapper.internal.asm.commons.SerialVersionUIDAdder.computeSHAdigest(byte[]):byte[]:427
Calling Method: org.modelmapper.internal.asm.commons.SerialVersionUIDAdder.computeSHAdigest(byte[]):byte[]
Source File: org\modelmapper\internal\asm\commons\SerialVersionUIDAdder.java
CWE: 327
API: java.security.MessageDigest.getInstance(String):MessageDigest
Caller: org.modelmapper.internal.asm.commons.SerialVersionUIDAdder.computeSHAdigest(byte[]):byte[]:427
Call: getInstance("SHA")
```

Is it possible to change this code so that it does not use an insecure algorithm (SHA) to create a message digest?  Thank you!

**Comment by mauricio-grando** [comment]
Hello @troy-bishop. Did you solved this scanning problem?

I'm working with HCL AppScan and got same situation here, without any solution yet =(

**Comment by troy-bishop** [comment]
No, but I looked into it more and we marked it as a false-positive with the following explanation:

This vulnerability is found in a dependency of modelmapper, coming from ASM (https://asm.ow2.io/), and they document that SHA-1 is used for computing the serialVersionUID based on Oracles specification: https://gitlab.ow2.org/asm/asm/-/blob/master/asm-commons/src/main/java/org/objectweb/asm/commons/SerialVersionUIDAdder.java#L56 Looking at Oracle's spec (from Java 18, the latest): https://docs.oracle.com/en/java/javase/18/docs/specs/serialization/class.html#stream-unique-identifiers "The serialVersionUID is computed using the signature of a stream of bytes that reflect the class definition. The National Institute of Standards and Technology (NIST) Secure Hash Algorithm (SHA-1) is used to compute a signature for the stream."

If it's not considered a security vulnerability in Java then it is not in this case either.
